osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -85,50 +82,49 @@ public abstract partial class SongSelect : ScreenWithBeatmapBackground, IKeyBind
         public bool BeatmapSetsLoaded => IsLoaded && Carousel?.BeatmapSetsLoaded == true;
 
         [Resolved]
-        private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; }
+        private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; } = null!;
 
-        protected BeatmapCarousel Carousel { get; private set; }
+        protected BeatmapCarousel? Carousel { get; private set; }
```

Comment:
Not sure how to feel about marking beatmap carousel as nullable, it's guaranteed not to be null anytime after loading the screen. We usually suppress null in that case similar to other components initialized on BDL and resolved dependencies.

That being said, special consideration needs to be taken on any usage of the screen before load (such as `BeatmapSetsLoaded`, potentially)

Reply:
yeah this was irking me ( the fact that it was guaranteed not to be null most of the  time) but i didn't know how to address it without making it nullable, ill go change it